src/ActionKit/CsrfTokenProvider.php

```diff
@@ -0,0 +1,65 @@
+<?php
+namespace ActionKit;
+use ActionKit\CsrfToken;
+use Exception;
+
+class CsrfTokenProvider {
+
+    public $tokenSessionId = '_csrf_token';
+    public $timeout = 300;
+
+    public function __construct($tokenSessionId = '_csrf_token', $timeout=300){
+        $this->tokenSessionId = $tokenSessionId;
+        $this->timeout = $timeout;
+    }
+
+    public function generateToken() {
+        $token = new CsrfToken($this->tokenSessionId, $this->timeout);
+        $token->time = time();
+        $token->salt = $this->randomString(32);
+        $token->sessid = session_id();
+        $token->ip = $_SERVER['REMOTE_ADDR'];
+
+        $_SESSION[$token->tokenSessionId] = serialize($token);
+
+        $hash = $this->calculateHash($token);
+        return base64_encode($hash);
+    }
+
+    public function checkToken($hashCsrfToken) {
```

Comment:
This method is to verify the hash we got and the hash exists in our session. hence, this method requires two arguments:  1. token in the current session   2. token hash to verify

```php
static public function verifyToken(CsrfToken $token, $tokenHash) {
   // ....
}
```

The use case:

```php
$token = CsrfTokenProvider::loadTokenWithSessionKey('_csrf_token');
if (CsrfTokenProvider::verifyToken($token, $hash)) {
    // valid token
} else {
   // invalid
}
```